app-modules/portal/src/DataTransferObjects/KnowledgeBaseCategoryData.php

```diff
@@ -45,5 +47,7 @@ public function __construct(
         public string $name,
         public ?string $description,
         public ?string $icon,
+        public ?Model $parentCategory,
```

Comment:
You should be able to type hint the actual Model of Knowledge Category rather than the base Model. Though you may want to consider making this another `KnowledgeBaseCategoryData` so that you only pass it the data needed rather than the entire Model.